<bug>
          <bug_id>5590</bug_id>
          
          <creation_ts>2011-09-07 09:44:00 +0200</creation_ts>
          <short_desc>When the disposal of an object fails, we get no stack trace</short_desc>
          <delta_ts>2014-03-03 15:23:55 +0100</delta_ts>
          <reporter_accessible>1</reporter_accessible>
          <cclist_accessible>1</cclist_accessible>
          <classification_id>1</classification_id>
          <classification>Unclassified</classification>
          <product>framework</product>
          <component>core</component>
          <version>master</version>
          <rep_platform>All</rep_platform>
          <op_sys>All</op_sys>
          <bug_status>CLOSED</bug_status>
          <resolution>FIXED</resolution>
          
          
          
          
          
          <priority>P4</priority>
          <bug_severity>enhancement</bug_severity>
          <target_milestone>1.6</target_milestone>
          
          
          
          <everconfirmed>1</everconfirmed>
          <reporter name="Uwe Stahl">uwe.stahl</reporter>
          <assigned_to name="Alexander Steitz">alexander.steitz</assigned_to>
          
          <qa_contact name="Alexander Steitz">alexander.steitz</qa_contact>
          <cf_browser>---</cf_browser>
          

      

      

      

          <long_desc isprivate="0">
            <commentid>26693</commentid>
            <who name="Uwe Stahl">uwe.stahl</who>
            <bug_when>2011-09-07 09:44:31 +0200</bug_when>
            <thetext>Our error logging database contains some error messages that state that the disposal of some objects have failed.

The problem is that the error message only says the which class could not be disposed, but the error message does not contain the exact stack trace of the error.
The destruct methods of our classes contain complex code, and the destruct method itself calls other methods. Therefore we need this stack trace.

I would recommend to add the stack trace by changing the following line in the class qx.core.ObjectRegistry 
from:
   qx.Bootstrap.error(this, &quot;Could not dispose object &quot; + obj.toString() + &quot;: &quot; + ex);
either to:
    qx.Bootstrap.error(this, &quot;Could not dispose object &quot; + obj.toString() + &quot;: &quot; + ex, ex);
or to: 
    qx.Bootstrap.error(this, &quot;Could not dispose object &quot; + obj.toString() + &quot;: &quot; + ex + &quot;: &quot; + ex.getStackTrace() );

This change should generate the stack trace that we need.</thetext>
          </long_desc>
          <long_desc isprivate="0">
            <commentid>26715</commentid>
            <who name="Martin Wittemann">martin.wittemann</who>
            <bug_when>2011-09-07 17:37:23 +0200</bug_when>
            <thetext>Alex, please take a look at this. Also keep in mind to mark the stacktrace class, if necessary, as optional. Don&#39;t want to have another dependency in the ObjectRegistry.</thetext>
          </long_desc>
          <long_desc isprivate="0">
            <commentid>26859</commentid>
            <who name="Alexander Steitz">alexander.steitz</who>
            <bug_when>2011-09-21 11:48:40 +0200</bug_when>
            <thetext>Fixed issue with revision #28711 (trunk)</thetext>
          </long_desc>
          <long_desc isprivate="0">
            <commentid>26860</commentid>
            <who name="Alexander Steitz">alexander.steitz</who>
            <bug_when>2011-09-21 11:49:34 +0200</bug_when>
            <thetext>@Martin
We do not need &quot;qx.dev.StackTrace&quot; functionality here, since the Logger already serializes an error object (showing the stack trace).</thetext>
          </long_desc>
          <long_desc isprivate="0">
            <commentid>38196</commentid>
            <who name="Martin Wittemann">martin.wittemann</who>
            <bug_when>2014-03-03 15:23:55 +0100</bug_when>
            <thetext>Closed all bugs already shipped with a release.</thetext>
          </long_desc>
      
      

    </bug>